keps/sig-node/4216-image-pull-per-runtime-class/README.md

```diff
@@ -0,0 +1,807 @@
+Additionally, having support to pull images per runtime class might also help projects like supporting different snapshotters with different runtimes as well.
+
+### Goals
+
+This KEP describes the detailed design and API changes needed in kubelet and CRI to support image pull with optional runtime class option.
+Goal is to keep the functionality same as it is today if user does not use the runtime class option with image pull.
+
+### Non-Goals
+
+This KEP only addresses the changes needed on kubelet and CRI. It does not go over the changes needed on container runtimes like containerd, CRI-O etc to support this feature. The implementation details should be handled by the respective container runtimes.
+
+Also, this KEP does not intend to address snapshotter per runtime class. It is different from image pull per runtime class and would need a separate KEP to address it.
+
+## Proposal
+
+## Design Details
+
+### CRI/kubelet changes:
+
+With this feature, multiple versions of the same image can exist on a single node.
+For example, python image can exist on the node for WS2022 (OSVersion = 20348) and WS2019(OSVersion = 17703) depending on the guestOSVersion mentioned in containerd's toml file for a runtime.
+Therefore, it is important for kubelet, containerd, and other components to identify the image as its name and runtimeClass rather than just the image name.
+The following sections details the major changes that would need to be made to the different components to support this.
+
+#### ImageSpec
+Include field on CRI `ImageSpec` message to specify optional runtime class to use to pull image
+
+
+**Proposed:**
+
+<pre>
+// ImageSpec is an internal representation of an image.
+message ImageSpec {
+    // Container's Image field (e.g. imageID or imageDigest).
+    string image = 1;
+    // Unstructured key-value map holding arbitrary metadata.
+    // ImageSpec Annotations can be used to help the runtime target specific
+    // images in multi-arch images.
+    map<string, string> annotations = 2;
+    // The container image reference specified by the user (e.g. image[:tag] or digest).
+    // Only set if available within the RPC context.
+    string user_specified_image = 18;
+    // Named runtime configuration to use for pulling the image.
+    // If the runtime handler is unknown, this request should be rejected.  An
+    // empty string should select the default handler, equivalent to the
+    // behavior before this feature was added.
+    <mark>string runtime_handler = 19;</mark>
+}
+</pre>
+
+The "RuntimeClassName" of [podspec](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L3084) can be used to specify the runtime class name and then can be populated in ImageSpec struct during **EnsureImageExists()** call in kubelet.
+
+ImageSpec is referenced in the following APIs and can set the RuntimeHandler
+field if needed:
+
+1. PullImageRequest
+2. RemoveImageRequest
+3. ImageStatusRequest
+
+### Kubelet changes
```

Comment:
Can you specify the name of the feature-gate that will be used to control having the kubelet extra information described below here and also update it in the kep.yaml file?